landoapi/models/landing_job.py

```diff
@@ -272,6 +275,20 @@ def set_landed_revision_diffs(self):
                 .values(diff_id=revision.diff_id)
             )
 
+    def set_landed_reviewers(self, path: Path):
+        """Set approving peers and owners at time of landing."""
+        directory = Directory(FileConfig(path))
+        for revision in self.revisions:
+            approved_by = revision.data.get("approved_by")
+            if not approved_by:
+                continue
+            data = {"peers_or_owners": []}
+            for reviewer in approved_by:
+                if reviewer in directory.peers_and_owners:
+                    data["peers_or_owners"].append(reviewer)
+            data.update(revision.data)
```

Comment:
If `revision.data` already included a `peers_or_owners` field, wouldn't this overwrite it? Maybe we want to `copy.deepcopy(revision.data)` and then add the `peers_or_owners` key instead?

Reply:
Good point, this could happen for revisions that land more than once. I've reworked the way this is implemented to avoid having to create a new dictionary (which was one way to trigger a "modified" state on the JSONB field.).